omod-2.0/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/converter/openmrs2_0/AllergyReactionConverter2_0.java

```diff
@@ -0,0 +1,106 @@
+ * Version 1.0 (the "License"); you may not use this file except in
+ * compliance with the License. You may obtain a copy of the License at
+ * http://license.openmrs.org
+ *
+ * Software distributed under the License is distributed on an "AS IS"
+ * basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See the
+ * License for the specific language governing rights and limitations
+ * under the License.
+ *
+ * Copyright (C) OpenMRS, LLC.  All Rights Reserved.
+ */
+package org.openmrs.module.webservices.rest.web.v1_0.converter.openmrs2_0;
+
+import org.openmrs.Allergy;
+import org.openmrs.AllergyReaction;
+import org.openmrs.Concept;
+import org.openmrs.annotation.Handler;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.webservices.rest.SimpleObject;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.api.RestService;
+import org.openmrs.module.webservices.rest.web.representation.Representation;
+import org.openmrs.module.webservices.rest.web.resource.api.Converter;
+import org.openmrs.module.webservices.rest.web.response.ConversionException;
+import org.openmrs.module.webservices.rest.web.v1_0.resource.openmrs1_11.ConceptResource1_11;
+import org.openmrs.module.webservices.rest.web.v1_0.resource.openmrs2_0.PatientAllergyResource2_0;
+import java.util.LinkedHashMap;
+
+/**
+ * An implementation of Converter to be able to create a representation from a User
+ * when User is used in another resource. Currently UserAndPassword doesn't convert User
+ */
+@Handler(supports = AllergyReaction.class, order = 0)
+public class AllergyReactionConverter2_0 implements Converter<AllergyReaction> {
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.api.Converter#getByUniqueId(java.lang.String)
+	 */
+	@Override
+	public AllergyReaction getByUniqueId(String string) {
+		//return Context.getUserService().getUserByUuid(string);
+		return null;
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.api.Converter#newInstance(java.lang.String)
+	 */
+	@Override
+	public AllergyReaction newInstance(String type) {
+		return new AllergyReaction();
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.api.Converter#asRepresentation(T, org.openmrs.module.webservices.rest.web.representation.Representation)
+	 */
+	@Override
+	public SimpleObject asRepresentation(AllergyReaction instance, Representation rep) throws ConversionException {
+		SimpleObject allergenReactionObject = new SimpleObject();
+		allergenReactionObject.add("allergyReactionId", instance.getAllergyReactionId());
+		//PatientAllergyResource2_0 allergyResource = (PatientAllergyResource2_0) Context.getService(RestService.class).getResourceBySupportedClass(Allergy.class);
+		//allergenReactionObject.add("allergy", allergyResource.asRepresentation(instance.getAllergy(), rep));
+		allergenReactionObject.add("allergy", instance.getAllergy().getAllergyId());
```

Comment:
I don't see why you would need to include a link to the containing object here since you can only get this AllergyReaction inside an Allergy. (Also, don't use the primary key id.)